src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs

```diff
@@ -0,0 +1,128 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Logging;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Represents a generic configuration manager.
+    /// </summary>
+    public abstract class BaseConfigurationManager
+    {
+        private TimeSpan _automaticRefreshInterval = DefaultAutomaticRefreshInterval;
+        private TimeSpan _refreshInterval = DefaultRefreshInterval;
+
+        /// <summary>
+        /// Gets or sets the <see cref="TimeSpan"/> that controls how often an automatic metadata refresh should occur.
+        /// </summary>
+        public TimeSpan AutomaticRefreshInterval
+        {
+            get { return _automaticRefreshInterval; }
+            set
+            {
+                if (value < MinimumAutomaticRefreshInterval)
+                    throw LogHelper.LogExceptionMessage(new ArgumentOutOfRangeException(nameof(value), LogHelper.FormatInvariant(LogMessages.IDX10107, MinimumAutomaticRefreshInterval, value)));
+
+                _automaticRefreshInterval = value;
+            }
+        }
+
+        /// <summary>
+        /// The most recently retrieved configuration.
+        /// </summary>
+        public BaseConfiguration CurrentConfiguration { get; set; }
+
+        /// <summary>
+        /// 12 hours is the default time interval that afterwards, <see cref="GetBaseConfigurationAsync(CancellationToken)"/> will obtain new configuration.
+        /// </summary>
+        public static readonly TimeSpan DefaultAutomaticRefreshInterval = new TimeSpan(0, 12, 0, 0);
+
+        /// <summary>
+        /// 5 minutes is the default time interval that must pass for <see cref="RequestRefresh"/> to obtain a new configuration.
+        /// </summary>
+        public static readonly TimeSpan DefaultRefreshInterval = new TimeSpan(0, 0, 5, 0);
+
+        /// <summary>
+        /// The last known good configuration (a configuration retrieved in the past that we were able to successfully validate a token against).
+        /// </summary>
+        public BaseConfiguration LKGConfiguration { get; set; }
+
+
```

Comment:
nit: extra new line?

Reply:
I think this was fixed in the first LKG PR that was recently merged in. :)